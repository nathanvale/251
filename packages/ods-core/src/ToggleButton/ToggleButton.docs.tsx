import { ComponentDocs } from "@origin-digital/ods-types";
import { ToggleButtonProps } from "./ToggleButton";
import { generateToggleButton } from "./ToggleButton.helper";

export const docs: ComponentDocs<ToggleButtonProps> = {
  category: "Interaction",
  componentName: "ToggleButton",
  description: [
    "This component has similar functionality as a Radio.",
    "This is rarely used by itself and almost always should be used as a child of a ToggleButtonGroup.",
  ].join(" "),
  propDescriptions: {
    disabled: `If true wil render a disabled button`,
    "aria-label": [
      `A screen-reader friendly label for this button. If is not provided the internal ToggleButton`,
      `sets it to "value" prop`,
    ].join(" "),
    children: [
      `The main content that is rendered inside the ToggleButton.`,
      `In most cases it is just a string. But it is not limited to that and it could be any valid`,
      `React element such as Icons.`,
    ].join(" "),
    "data-id": `The id to be used for test purposes. By default is set to the "value" prop`,
    selected: [
      `Can be used to define the "selected state of the component. `,
      `If selected is true the button is rendered as visually selected.`,
    ].join(" "),
    value: [
      `This defines the accepted values that ToggleButtonGroup's value prop can have.`,
      `It always has to be set.`,
    ].join(" "),
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => generateToggleButton(),
    },
    additional: [
      {
        label: "Selected",
        Code: () => generateToggleButton({ selected: true }),
      },
      {
        label: "Disabled",
        Code: () => generateToggleButton({ disabled: true }),
      },
    ],
  },
  snippets: [
    {
      label: "Default",
      Code: () => generateToggleButton(),
    },
  ],
};
