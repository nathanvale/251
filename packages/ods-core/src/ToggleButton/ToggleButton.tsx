import React from "react";
import {
  ToggleButton as MuiToggleButton,
  ToggleButtonProps as MuiToggleButtonProps,
} from "@material-ui/lab";
import { ComponentBaseProps, MuiProps } from "@origin-digital/ods-types";
import { fade, makeStyles } from "@material-ui/core/styles";

/**
 * selected toggle button:
 *   BG: rgba(255, 180, 50, 0.08) -> fade(secondary, 0.08)
 *   border: rgb(255, 180, 50) -> secondary?
 * selected hover:
 *   BG: rgba(255, 180, 50, 0.12) -> fade(secondary, 0.12)
 *   border: rgba(255, 180, 50) -> secondary?
 * not selected hover:
 *   BG: rgba(80, 80, 80, 0.04)
 *   border: rgba(80,80,80, 0.16)
 *
 * not selected disabled:
 *   BG: transparent
 *   color: rgba(80,80,80, 0.16)
 *   border: rgba(80,80,80, 0.16) (unchanged by disabled)
 *
 * selected disabled:
 *   BG: rgba(255, 180, 50, 0.08) (unchanged by disabled)
 */

export interface ToggleButtonProps
  extends Omit<ComponentBaseProps, "children"> {
  children: React.ReactNode;
  value: string | number;
  "aria-label"?: string;
  muiProps?: MuiProps<MuiToggleButtonProps>;
  selected?: boolean;
}

const useToggleButtonStyles = makeStyles(
  (theme) => ({
    root: {
      fontSize: theme.typography.body1.fontSize,
      border: `solid 1px ${theme.palette.grey[200]}`,
      color: theme.palette.grey[500],
      "&:hover": {
        backgroundColor: fade(theme.palette.grey[500], 0.04),
      },
      "&$selected$selected": {
        backgroundColor: fade(theme.palette.secondary.main, 0.08),
        border: `solid 1px ${theme.palette.secondary.main}`,
        "& .MuiToggleButton-label": {
          fontWeight: theme.typography.fontWeightMedium,
        },
        "&:hover": {
          backgroundColor: fade(theme.palette.secondary.main, 0.12),
          border: `solid 1px ${theme.palette.secondary.main}`,
        },
      },
      "&$disabled": {
        color: theme.palette.grey[200],
      },
    },
    selected: {},
    disabled: {},
  }),
  { classNamePrefix: "toggle-button" }
);

export const ToggleButton = ({
  "aria-label": ariaLabel,
  children,
  classes,
  "data-id": dataId,
  muiProps,
  value,
  ...props
}: ToggleButtonProps) => {
  const calcAriaLabel =
    ariaLabel || (typeof children === "string" && children) || String(value);

  const calcDataId = dataId || value;

  const tbStyleClasses = useToggleButtonStyles();
  return (
    <MuiToggleButton
      aria-label={calcAriaLabel}
      classes={{ ...tbStyleClasses, ...(classes || {}) }}
      data-id={calcDataId}
      value={value}
      {...props}
      {...(muiProps || {})}
    >
      {children}
    </MuiToggleButton>
  );
};

ToggleButton.defaultProps = {};

ToggleButton.displayName = "ToggleButton";
