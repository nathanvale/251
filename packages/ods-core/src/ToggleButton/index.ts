export { ToggleButton } from "./ToggleButton";
export type ToggleButtonProps = import("./ToggleButton").ToggleButtonProps;
