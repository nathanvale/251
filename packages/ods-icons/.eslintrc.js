module.exports = {
  overrides: [
    {
      files: ["src/mui/!(index)*.ts", "src/custom/!(index)*.tsx"],
      rules: {
        "import/newline-after-import": "off",
        "import/no-default-export": "off",
      },
    },
  ],
};
