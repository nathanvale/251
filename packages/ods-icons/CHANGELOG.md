# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.13.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.12.0...@origin-digital/ods-icons@0.13.0) (2020-08-05)


### Features

* **ods-core:** Add IconButton component ([a067bbc](https://bitbucket.orgn.io/od/origin-ui/commits/a067bbcdc974015ef255c7a998ffb1f119ba76bd))





# [0.12.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.13...@origin-digital/ods-icons@0.12.0) (2020-07-29)


### Features

* **alert:** implement alert ([7486c7d](https://bitbucket.orgn.io/od/origin-ui/commits/7486c7d265b558bb0673c63fc5d863d330db0cbf))





## [0.11.13](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.12...@origin-digital/ods-icons@0.11.13) (2020-07-28)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.11.12](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.11...@origin-digital/ods-icons@0.11.12) (2020-07-22)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.11.11](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.10...@origin-digital/ods-icons@0.11.11) (2020-07-20)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.11.10](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.9...@origin-digital/ods-icons@0.11.10) (2020-07-16)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.11.9](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.8...@origin-digital/ods-icons@0.11.9) (2020-07-15)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.11.8](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.7...@origin-digital/ods-icons@0.11.8) (2020-07-13)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.11.7](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.6...@origin-digital/ods-icons@0.11.7) (2020-07-13)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.11.6](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.5...@origin-digital/ods-icons@0.11.6) (2020-07-09)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.11.5](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.4...@origin-digital/ods-icons@0.11.5) (2020-07-06)


### Bug Fixes

* **checkbox:** centre it vertically ([c09e0a3](https://bitbucket.orgn.io/od/origin-ui/commits/c09e0a37e954b9252d6e0a6fb1a34470af44393f))





## [0.11.4](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.3...@origin-digital/ods-icons@0.11.4) (2020-07-03)


### Bug Fixes

* **data-id:** default it to id ([ea5cf55](https://bitbucket.orgn.io/od/origin-ui/commits/ea5cf55ef5a342e2e1310a412d68340ef9284afd))





## [0.11.3](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.2...@origin-digital/ods-icons@0.11.3) (2020-07-01)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.11.2](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.1...@origin-digital/ods-icons@0.11.2) (2020-06-30)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.11.1](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.11.0...@origin-digital/ods-icons@0.11.1) (2020-06-29)

**Note:** Version bump only for package @origin-digital/ods-icons





# [0.11.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.10.4...@origin-digital/ods-icons@0.11.0) (2020-06-23)


### Features

* implement icons ([89f8fab](https://bitbucket.orgn.io/od/origin-ui/commits/89f8fabe0e8ee79a0efd58d835281285845feca5))





## [0.10.4](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.10.3...@origin-digital/ods-icons@0.10.4) (2020-06-15)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.10.3](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.10.2...@origin-digital/ods-icons@0.10.3) (2020-06-15)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.10.2](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.10.1...@origin-digital/ods-icons@0.10.2) (2020-06-15)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.10.1](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.10.0...@origin-digital/ods-icons@0.10.1) (2020-06-09)

**Note:** Version bump only for package @origin-digital/ods-icons





# [0.10.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.9.0...@origin-digital/ods-icons@0.10.0) (2020-06-05)


### Features

* **textfieldbase:** create a base for several field components ([25a4b90](https://bitbucket.orgn.io/od/origin-ui/commits/25a4b901c75689a80eaf29e116e55e6c5b0c6991))





# [0.9.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.8.3...@origin-digital/ods-icons@0.9.0) (2020-05-27)


### Features

* **OriginThemeProvider:** allow configurable link type ([5a39d9d](https://bitbucket.orgn.io/od/origin-ui/commits/5a39d9dd68eb4fe19b7b80c071f020ab73b5df61))





## [0.8.3](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.8.2...@origin-digital/ods-icons@0.8.3) (2020-05-21)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.8.2](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.8.1...@origin-digital/ods-icons@0.8.2) (2020-05-20)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.8.1](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.8.0...@origin-digital/ods-icons@0.8.1) (2020-05-18)

**Note:** Version bump only for package @origin-digital/ods-icons





# [0.8.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.7.4...@origin-digital/ods-icons@0.8.0) (2020-05-15)


### Features

* uplift docs ([c6ba10f](https://bitbucket.orgn.io/od/origin-ui/commits/c6ba10fdadaf6a8640ea2473ba5facfa9f2c2123))





## [0.7.4](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.7.3...@origin-digital/ods-icons@0.7.4) (2020-05-08)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.7.3](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.7.2...@origin-digital/ods-icons@0.7.3) (2020-05-05)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.7.2](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.7.1...@origin-digital/ods-icons@0.7.2) (2020-05-01)


### Bug Fixes

* **docs:** icons are not portable from docs snippets to playroom ([0e57a26](https://bitbucket.orgn.io/od/origin-ui/commits/0e57a2602797dc3ecccc67c86b101d52506ff7dc))





## [0.7.1](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.7.0...@origin-digital/ods-icons@0.7.1) (2020-04-30)

**Note:** Version bump only for package @origin-digital/ods-icons





# [0.7.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.6.4...@origin-digital/ods-icons@0.7.0) (2020-04-23)


### Features

* implement ChevronLink, ChevronButton ([80f471d](https://bitbucket.orgn.io/od/origin-ui/commits/80f471dfdf21600c8526b9b681fae8e5c18d8046))





## [0.6.4](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.6.3...@origin-digital/ods-icons@0.6.4) (2020-04-20)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.6.3](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.6.2...@origin-digital/ods-icons@0.6.3) (2020-04-20)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.6.2](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.6.1...@origin-digital/ods-icons@0.6.2) (2020-04-08)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.6.1](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.6.0...@origin-digital/ods-icons@0.6.1) (2020-04-06)

**Note:** Version bump only for package @origin-digital/ods-icons





# [0.6.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.12...@origin-digital/ods-icons@0.6.0) (2020-03-31)


### Features

* implement ods master color palette ([682e0dd](https://bitbucket.orgn.io/od/origin-ui/commits/682e0dd33cfd270cb51d0e720a0ab59bdfda2f87))





## [0.5.12](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.11...@origin-digital/ods-icons@0.5.12) (2020-03-25)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.5.11](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.10...@origin-digital/ods-icons@0.5.11) (2020-03-25)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.5.10](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.9...@origin-digital/ods-icons@0.5.10) (2020-03-19)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.5.9](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.8...@origin-digital/ods-icons@0.5.9) (2020-03-16)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.5.8](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.7...@origin-digital/ods-icons@0.5.8) (2020-03-13)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.5.7](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.6...@origin-digital/ods-icons@0.5.7) (2020-03-11)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.5.6](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.5...@origin-digital/ods-icons@0.5.6) (2020-03-10)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.5.5](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.4...@origin-digital/ods-icons@0.5.5) (2020-03-06)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.5.4](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.3...@origin-digital/ods-icons@0.5.4) (2020-03-06)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.5.3](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.2...@origin-digital/ods-icons@0.5.3) (2020-03-04)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.5.2](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.1...@origin-digital/ods-icons@0.5.2) (2020-02-27)

**Note:** Version bump only for package @origin-digital/ods-icons





## [0.5.1](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.0...@origin-digital/ods-icons@0.5.1) (2020-02-26)

**Note:** Version bump only for package @origin-digital/ods-icons





# [0.5.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.4.2...@origin-digital/ods-icons@0.5.0) (2020-02-19)


### Features

* dynamically generate docs platform ([0effbb8](https://bitbucket.orgn.io/od/origin-ui/commits/0effbb8))





## [0.4.2](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.5.0...@origin-digital/ods-icons@0.4.2) (2020-02-17)

**Note:** Version bump only for package @origin-digital/ods-icons

## [0.4.1](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.4.0...@origin-digital/ods-icons@0.4.1) (2020-02-12)

**Note:** Version bump only for package @origin-digital/ods-icons

# [0.4.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.3.3...@origin-digital/ods-icons@0.4.0) (2020-02-12)

### Features

- **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))

## [0.3.3](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.3.2...@origin-digital/ods-icons@0.3.3) (2020-02-10)

**Note:** Version bump only for package @origin-digital/ods-icons

## [0.3.2](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.3.1...@origin-digital/ods-icons@0.3.2) (2020-02-06)

**Note:** Version bump only for package @origin-digital/ods-icons

## [0.3.1](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.3.0...@origin-digital/ods-icons@0.3.1) (2020-02-05)

**Note:** Version bump only for package @origin-digital/ods-icons

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.2.0...@origin-digital/ods-icons@0.3.0) (2020-01-31)

### Features

- **stack, columns:** add alignment to allow fine tuning of spaces ([a3f0eed](https://bitbucket.orgn.io/od/origin-ui/commits/a3f0eed))

# [0.2.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.1.2...@origin-digital/ods-icons@0.2.0) (2020-01-31)

### Features

- restructure packages for theming, debugging and an incubator lab ([8c8a514](https://bitbucket.orgn.io/od/origin-ui/commits/8c8a514))

## [0.1.2](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-icons@0.1.1...@origin-digital/ods-icons@0.1.2) (2020-01-23)

**Note:** Version bump only for package @origin-digital/ods-icons

## 0.1.1 (2020-01-22)

**Note:** Version bump only for package @origin-digital/ods-icons
