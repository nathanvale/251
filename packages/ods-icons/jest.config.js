const base = require("../../jest.settings.js");
const pack = require("./package.json");

module.exports = {
  ...base,
  coveragePathIgnorePatterns: ["/ods-icons/src/custom/", "/ods-icons/src/mui/"],
  displayName: pack.name,
  name: pack.name,
  globals: {
    __DEV__: true,
  },
};
