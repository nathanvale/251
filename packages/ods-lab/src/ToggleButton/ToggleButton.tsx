import * as React from "react";
import MuiToggleButton, {
  ToggleButtonProps as IToggleButton,
} from "@material-ui/lab/ToggleButton";

export interface ToggleButtonProps extends IToggleButton {}

export const ToggleButton = (props: ToggleButtonProps) => (
  <MuiToggleButton {...props}>{props.children}</MuiToggleButton>
);
