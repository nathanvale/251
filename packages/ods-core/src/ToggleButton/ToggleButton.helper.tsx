import React from "react";
import { ToggleButton, ToggleButtonProps } from "./ToggleButton";

const defaultProps: ToggleButtonProps = {
  children: "Melbourne",
  value: "mlb",
};

export const generateToggleButton = (
  props: Partial<ToggleButtonProps> = {}
) => <ToggleButton {...defaultProps} {...props} />;
