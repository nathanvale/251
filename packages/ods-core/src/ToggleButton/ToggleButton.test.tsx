import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { IconElectricity } from "@origin-digital/ods-icons";
import { ToggleButtonProps } from "./ToggleButton";
import { generateToggleButton } from "./ToggleButton.helper";

const renderToggleButton = (
  toggleButtonProps: Partial<ToggleButtonProps> = {}
) => render(generateToggleButton(toggleButtonProps));

test("It renders", () => {
  const { container } = renderToggleButton();
  expect(container).toMatchSnapshot();
});

test("It renders in selected state", () => {
  const { container } = renderToggleButton({ selected: true });
  expect(container).toMatchSnapshot();
});

test("It can render icon", () => {
  const { container } = renderToggleButton({ children: <IconElectricity /> });
  expect(container).toMatchSnapshot();
});
