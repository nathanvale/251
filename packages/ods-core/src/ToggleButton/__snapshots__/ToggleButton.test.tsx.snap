// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It can render icon 1`] = `
<div>
  <button
    aria-label="mlb"
    class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-7"
    data-id="mlb"
    tabindex="0"
    type="button"
    value="mlb"
  >
    <span
      class="MuiToggleButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root svg-icon-secondary-small-svg-icon-10"
        data-id="IconElectricity"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9 21c0 .55.45 1 1 1h4c.55 0 1-.45 1-1v-1H9v1zm3-19C8.14 2 5 5.14 5 9c0 2.38 1.19 4.47 3 5.74V17c0 .55.45 1 1 1h6c.55 0 1-.45 1-1v-2.26c1.81-1.27 3-3.36 3-5.74 0-3.86-3.14-7-7-7zm2.85 11.1l-.85.6V16h-4v-2.3l-.85-.6A4.997 4.997 0 0 1 7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 1.63-.8 3.16-2.15 4.1z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`It renders 1`] = `
<div>
  <button
    aria-label="Melbourne"
    class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-1"
    data-id="mlb"
    tabindex="0"
    type="button"
    value="mlb"
  >
    <span
      class="MuiToggleButton-label"
    >
      Melbourne
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`It renders in selected state 1`] = `
<div>
  <button
    aria-label="Melbourne"
    aria-pressed="true"
    class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-4 Mui-selected toggle-button-selected-5"
    data-id="mlb"
    tabindex="0"
    type="button"
    value="mlb"
  >
    <span
      class="MuiToggleButton-label"
    >
      Melbourne
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
